internal/api/core/response.go

```diff
@@ -70,3 +71,56 @@ func getRequestInfo(request ValidAPIRequest) (string, timestamp.Timestamp) {
 
 	return id, startTime
 }
+
+// Get the endpoint, userEmail, courseID, assignmentID, and locator
+// from a ValidAPIRequest and an APIError, both of which may be nil.
+func getRequestInfoForStats(request ValidAPIRequest, apiError *APIError) (string, string, string, string, string) {
```

Comment:
I don't think the "ForStats" part of the name is necessary.
There's nothing here that is specific to stats.